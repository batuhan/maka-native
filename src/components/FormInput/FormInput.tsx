import React from "react";
import { StyleSheet, TextInput } from "react-native";

type Props = {
  error?: string;
  value?: string;
  onChangeText: (name: string, value: string) => void;
  style?: {};
  name: string;
  placeholder: string;
};

class FormInput extends React.Component<Props> {
  render() {
    //handle error
    const { style, error, value, onChangeText, name, placeholder } = this.props;
    return (
      <TextInput
        style={[styles.textInput, style]}
        value={value}
        placeholder={placeholder}
        onChangeText={(value: string) => onChangeText(name, value)}
      />
    );
  }
}

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    borderBottomWidth: StyleSheet.hairlineWidth,
    marginBottom: 20
  }
});

export default FormInput;
