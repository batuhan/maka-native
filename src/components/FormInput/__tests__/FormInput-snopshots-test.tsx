import React from "react";
import renderer from "react-test-renderer";

import FormInput from "../FormInput";

describe("FormInput", () => {
  describe("Rendering", () => {
    it("should match to snapshot", () => {
      const placeholder = "form input placeholder";
      const tree = renderer
        .create(<FormInput placeholder={placeholder} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
