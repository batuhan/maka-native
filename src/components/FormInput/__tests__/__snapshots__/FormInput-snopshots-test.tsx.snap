// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput Rendering should match to snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  onChangeText={[Function]}
  placeholder="form input placeholder"
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderBottomWidth": 0.5,
        "height": 40,
        "marginBottom": 20,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
